react-projects/src/codeImprovement.js

```diff
@@ -0,0 +1,52 @@
+//Exercise 1:
+const average=(numbers)=>numbers.reduce((acc,current)=>acc+current,0)/numbers.length
+console.log(average([1, 2]));
+
+//Exercise 2:
+
+const greet = (name) => name ? `Hello, ${name}!` : "Hello!"
+console.log(greet('divine'));
+
+//Exercise 3:
+const reverseString =(str)=>  str.split('').reverse().join('')
+console.log(reverseString('masdwe'));
+
+//Exercise 4:
+
+const findEvenNumbers = (numbers)=> numbers.filter((number)=> number %2 ===0)
+console.log(findEvenNumbers([1, 2, 3, 4, 5, 6]));
+
+//Exercise 5:
+const calculateFactorial=(n)=> {
+    if (n<1) return  1;
+        let factorial = 1;
+        for (let i = 2; i <= n; i++) {
+            factorial *= i;
+        }
+        return factorial;
+}
```

Comment:
On this exercise, you used many lines of code, as an improvement, you may use tenary operator  or any way to reduce lines of codes. Also on for loop , you may use for.. ... of